### ISSUE TITLE
[BUG] Missing datapoints in EMFLoggingMeterRegistry

### ISSUE THREAD

**Issue by chenqi0805** [opensearch-project/data-prepper#2469]
**Describe the bug**
when integrating with fluent-bit, there is missing metric data points occasionally.

**To Reproduce**
A cloudformation template:
```
AWSTemplateFormatVersion: "2010-09-09"
Description: "Template to deploy and monitor Data Prepper on ECS Fargate tasks"
Resources:
  FargateTask:
    Type: AWS::ECS::TaskDefinition
    Properties:
      NetworkMode: awsvpc
      Family: data-prepper-emf-metrics
      TaskRoleArn: arn:aws:iam::531516510575:role/ecsTaskExecutionRole
      ExecutionRoleArn: arn:aws:iam::531516510575:role/ecsTaskExecutionRole
      Cpu: 1024
      Memory: 8192
      RequiresCompatibilities:
        - "FARGATE"
      ContainerDefinitions:
        - Name: data-prepper-poc
          Image: 531516510575.dkr.ecr.us-east-1.amazonaws.com/data-prepper-poc:EMF-METRICS-LOGGING-POC
          MemoryReservation: 2048
          Essential: true
          PortMappings:
            - ContainerPort: 21890
              HostPort: 21890
              Protocol: tcp
            - ContainerPort: 2021
              HostPort: 2021
              Protocol: tcp
            - ContainerPort: 4900
              HostPort: 4900
              Protocol: tcp
          LogConfiguration:
            LogDriver: awsfirelens
            Options:
              Name: kinesis_streams
              Match: emf-DataPrepper
              log_key: log
              stream: MetricLogDataStream-dev-us-east-1
              region: us-east-1
          Command:
            - /bin/bash
            - -c
            - |
              echo $PIPELINES_YAML | base64 -d - | tee /usr/share/data-prepper/pipelines/pipelines.yaml
              echo $DATA_PREPPER_CONFIG_YAML | base64 -d - | tee /usr/share/data-prepper/config/data-prepper-config.yaml
              bin/data-prepper
          Environment:
            - Name: PIPELINES_YAML
              Value:
                Fn::Base64: |
                  log-pipeline:
                    source:
                      http:
                        health_check_service: true
                    processor:
                      - date:
                          from_time_received: true
                          destination: "@timestamp"
                    sink:
                      - stdout:
            - Name: DATA_PREPPER_CONFIG_YAML
              Value:
                Fn::Base64: |
                  ssl: false
                  metricRegistries:
                    - "EmbeddedMetricsFormat"
                    - "CloudWatch"
                  metricTags:
                    serviceName: test-emf
                    accountId: 531516510575
        - Name: log_router
          Image: public.ecr.aws/aws-observability/aws-for-fluent-bit:init-latest
          MemoryReservation: 50
          Essential: true
          FirelensConfiguration:
            Type: fluentbit
            Options:
              config-file-type: "file"
              config-file-value: "/fluent-bit.conf"
          LogConfiguration:
            LogDriver: awslogs
            Options:
              awslogs-group: 531516510575/test-log-pipeline-firelens-event-logs
              awslogs-region: us-east-1
              awslogs-stream-prefix: 531516510575/test-log-pipeline-firelens-event-logs
              awslogs-create-group: true
          Environment:
            - Name: FLUENT_BIT_CONF
              Value:
                Fn::Base64: |
                  [INPUT]
                    Name        tcp
                    Listen      0.0.0.0
                    Port        25888
                    Chunk_Size  32
                    Buffer_Size 64
                    Format      none
                    Tag         emf-DataPrepper
                  [FILTER]
                    Name                record_modifier
                    Match               data-prepper-poc*-firelens*
                    Record              resource_id 531516510575/test-log-pipeline
                  [OUTPUT]
                    Name                cloudwatch
                    Match               data-prepper-poc*-firelens*
                    region              us-east-1
                    log_key             log
                    log_group_name      pepper-poc-logs
                    log_stream_prefix   pepper-poc-logs
                    auto_create_group   true
                  [OUTPUT]
                    Name                cloudwatch
                    Match               emf-DataPrepper
                    region              us-east-1
                    log_key             log
                    log_group_name      pepper-metric-logs
                    log_stream_prefix   pepper-metric-logs
                    auto_create_group   true
                    log_format          json/emf
          Command:
            - /bin/bash
            - -c
            - |
              echo $FLUENT_BIT_CONF | base64 -d - | tee /fluent-bit.conf
              echo -n "AWS for Fluent Bit Container Image Version "
              cat /AWS_FOR_FLUENT_BIT_VERSION
              exec /fluent-bit/bin/fluent-bit -e /fluent-bit/firehose.so -e /fluent-bit/cloudwatch.so -e /fluent-bit/kinesis.so -c /fluent-bit/etc/fluent-bit.conf
```
deploy the template yields ECS task definition that one can create a service in an ECS cluster

**Expected behavior**
No missing data

**Screenshots**
![Screenshot 2023-04-10 at 11 34 15 AM](https://user-images.githubusercontent.com/19492223/230959731-09b57b3e-11a0-474d-8185-156a3e287c78.png)

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu 20.04 LTS]

**Additional context**

**Comment by chenqi0805** [comment]
> @chenqi0805 , Is Data Prepper logging any errors indicating any part of the problem here?

No error in data-prepper probably b/c the logs are at debug level in https://github.com/awslabs/aws-serverless-metrics-publisher

**Comment by chenqi0805** [comment]
Closed via #2467
